packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_proxy.dart

```diff
@@ -23,26 +36,19 @@ WKWebsiteDataStore _defaultWebsiteDataStore() =>
 /// it intends to return.
 class WebKitProxy {
   /// Constructs a [WebKitProxy].
-  const WebKitProxy({
-    this.createWebView = WKWebView.new,
+  WebKitProxy({
+    WebViewConstructor? createWebView,
```

Comment:
@bparrishMines Unfortunately I had to make this non-const because of the `Platform` switch. Better ideas welcome!

Reply:
You should be able to set this with a constant Function. Either by putting `(Platform.isIOS ? WKWebViewIOS.new : WKWebViewMacOS.new);` in a static method or a  top-level function.

I've also have been waiting for this feature! : https://github.com/dart-lang/language/issues/1048

Reply:
Thanks for the tip! I've re-added the consts I removed.

Unfortunately I had to duplicate all the arguments, because I couldn't make a static function that returned `Platform.isIOS ? WKWebViewIOS.new : WKWebViewMacOS.new` and then call that because it still didn't want me calling the function in the const constructor, so I had to instead make it a wrapper that calls the right constructor each time, passing all the args through. (If I missed a way to make that work please let me know!)